### DISCUSSION TITLE
Multi-version upgrade breaking changes?

### DISCUSSION THREAD

**Discussion by ascloutier** [josdejong/mathjs#2598] [Q&A]
In a recent security analysis, a company I work for realized that one of our internal libraries is still using mathjs v6.6.2 which has been flagged as having vulnerabilities. I have been tasked with upgrading and I'm trying to determine if there are any breaking changes that could impact that upgrade.

I recognize that we're _several_ version behind and I'm not asking for a comprehensive changelog. I'm really just trying to get a feel for whether any of the matrix operations have had breaking changes since 6.x. From what I can see, they have not but it seemed worthwhile to reach out to the community. 

We use the following mathjs functions:
Matrix class
Matrix.clone()
Matrix.size()

matrix()
add()
subtract()
multiply()
dot()
norm()
cross()
inv()
transpose()

I appreciate any insight that you could provide. Thank you.

**Comment by josdejong** [comment]
You can have a look at the descriptions of versions 7.0.0, 8.0.0, 9.0.0 and 10.0.0 in https://github.com/josdejong/mathjs/blob/develop/HISTORY.md.

It may help to upgrade one version at a time, there have been deprecation messages in place (which have been cleaned up again after a couple of versions)

**Reply by ascloutier** [reply]
Thank you. The only thing that looks like it might impact what we're using mathjs for is this one in 7.0.0:

_Improvements in calculation of the dot product of complex values. The first argument is now conjugated. See #1761. Thanks @m93a._

I will start upgrading and testing.
